### PR TITLE
Feat chatglm

### PR DESCRIPTION
**Pull request by better629**
feat
1. add `metagpt/provider/general_api_requestor.py` which finish general async api for http-based llm model 
2. add zhipuai api
3.  choose different LLM instance by `LLM()`, but need further to improve.

todo
1. improve the prompt to run the zhipuai successfully

zhipuai runs example and log
currently, it runs failed due to can't get prompt following output format.

[zhipuai_run_failed_log.log](https://github.com/geekan/MetaGPT/files/13400470/zhipuai_run_log.log)

[zhipuai_run_success_log.log](https://github.com/geekan/MetaGPT/files/13405395/zhipuai_run_success_log.log)
run the case successfully, but can't run.

### CODE REVIEW

**Comment by stellaHSR** [comment]
LGTM

**Comment by stellaHSR** [comment]
but seems conflict

File: metagpt/provider/zhipuai_api.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+from requests import ConnectionError
+
+import zhipuai
+
+from metagpt.config import CONFIG
+from metagpt.logs import logger
+from metagpt.provider.base_gpt_api import BaseGPTAPI
+from metagpt.provider.openai_api import CostManager, log_and_reraise
+from metagpt.provider.zhipuai.zhipu_model_api import ZhiPuModelAPI
+from metagpt.utils.ahttp_client import astream
```
Comment (shenchucheng) [inline metagpt/provider/zhipuai_api.py:23]: Module `metagpt.utils.ahttp_client` not found.

**Comment by shenchucheng** [comment]
LGTM

**Review by shenchucheng — APPROVED** [review]
LGTM

File: metagpt/config.py [hunk 2]
```diff
@@ -45,10 +45,11 @@ def __init__(self, yaml_file=default_yaml_file):
...
         self.anthropic_api_key = self._get("Anthropic_API_KEY")
-        if (not self.openai_api_key or "YOUR_API_KEY" == self.openai_api_key) and (
-                not self.anthropic_api_key or "YOUR_API_KEY" == self.anthropic_api_key
-        ):
-            raise NotConfiguredException("Set OPENAI_API_KEY or Anthropic_API_KEY first")
+        self.zhipuai_api_key = self._get("ZHIPUAI_API_KEY")
+        if (not self.openai_api_key or "YOUR_API_KEY" == self.openai_api_key) and \
+                (not self.anthropic_api_key or "YOUR_API_KEY" == self.anthropic_api_key) and \
+                (not self.zhipuai_api_key or "YOUR_API_KEY" == self.zhipuai_api_key):
+            raise NotConfiguredException("Set OPENAI_API_KEY or Anthropic_API_KEY or ZHIPUAI_API_KEY first")
```
[the remark this answers is not in the document]
Comment (better629) [reply metagpt/config.py:52]: will solve in the future together

File: metagpt/provider/zhipuai_api.py (hunk 1 above)
Comment (better629) [reply metagpt/provider/zhipuai_api.py]: forget to comment it, updated

**Pull request merged**